src/index.rs

```diff
@@ -134,15 +134,31 @@ impl<T> BitcoinCoreRpcResultExt<T> for Result<T, bitcoincore_rpc::Error> {
 impl Index {
   pub(crate) fn open(options: &Options) -> Result<Self> {
     let rpc_url = options.rpc_url();
+    let rpc_user = options.rpc_user();
+    let rpc_pass = options.rpc_pass();
     let cookie_file = options.cookie_file()?;
+    let cookie_file_path = cookie_file.clone();
 
-    log::info!(
-      "Connecting to Bitcoin Core RPC server at {rpc_url} using credentials from `{}`",
-      cookie_file.display()
-    );
 
+    // by default we use a cookie file
     let auth = Auth::CookieFile(cookie_file);
 
+    // if options.rpc_pass and rpc_user is not empty, use Auth::UserPass
+    let auth = if !rpc_pass.is_none() && !rpc_user.is_none() {
+      let ruser = rpc_user.unwrap();
+      let rpass = rpc_pass.unwrap();
+      log::info!("Using RPC credentials from command line {ruser}");
```

Comment:
`rpc_pass` and `rpc_user` are `Option<String>`, which you can get at in a nicer way than checking `! is_none()` then calling `unwrap()` (calling unwrap is generally (strongly) discouraged), by using [`if let`](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html) and [destructuring](https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_tuple.html):

```rust
let number = Some(7);
let another = Some(12);
let missing: Option<i32> = None;

if let (Some(i), Some(j)) = (number, another) {
    println!("Matched {:?} and {:?}!", i, j);
}
// ^ Prints "Matched 7 and 12!"

if let (Some(i), Some(j)) = (number, missing) {
    println!("Matched {:?} and {:?}!", i, j);
} else {
    println!("Didn't match");
}
// ^ Prints "Didn't match"
```

so something like

```suggestion
    if let (Some(rpass), Some(ruser))  = (rpc_pass, rpc_user) {
      log::info!("Using RPC credentials from command line {ruser}");
```

(Caveat I'm also a Rust noob so if the compiler complains, sorry!)